cmd/local.go

```diff
@@ -0,0 +1,367 @@
+		var err error
+
+		//Does it even exist?
+		if ok, _ := afero.Exists(fs, path.Join(pwi.FilesToClean.Location, v.File)); ok {
+			//Is it a directory?
+			if ok, _ := afero.IsDir(fs, path.Join(pwi.FilesToClean.Location, v.File)); ok {
+				err = fs.RemoveAll(path.Join(pwi.FilesToClean.Location, v.File))
+				//Nope it's a file!
+			} else {
+				err = fs.Remove(path.Join(pwi.FilesToClean.Location, v.File))
+			}
+
+			if err != nil {
+				//Indicate failure to remove
+				log.Printf("Failure removing file / directory %s. Details : %s", v.File, err.Error())
+			}
+		}
+	}
+
+	//Gitignore operations
+	createNewGitIgnoreFile(l.Nonmem)
+
+	//Mark as completed and move on to cleanup
+	channels.Completed <- 1
+}
+
+//End Scalable method definitions
+
+// runCmd represents the run command
+var localCmd = &cobra.Command{
+	Use:   "local",
+	Short: "local specifies to run a (set of) models locally",
+	Long:  runLongDescription,
+	Run:   local,
+}
+
+func init() {
+	runCmd.AddCommand(localCmd)
+}
+
+func local(cmd *cobra.Command, args []string) {
+
+	if debug {
+		viper.Debug()
+	}
+
+	if viper.ConfigFileUsed() != "" {
+		log.Printf("Config file loaded from %s", viper.ConfigFileUsed())
+	}
+
+	lo := localOperation{}
+
+	if verbose {
+		log.Printf("setting up a work queue with %v workers", viper.GetInt("threads"))
+	}
+
+	lo.Models = localModelsFromArguments(args)
+
+	if len(lo.Models) == 0 {
+		log.Fatal("No models were located or loaded. Please verify the arguments provided and try again")
+	}
+
+	//Display Summary
+
+	//Models Added
+	log.Printf("A total of %d models have been located for work", len(lo.Models))
+
+	//Models in Error
+	//Locate 'em
+	counter := 0
+	var errors []NonMemModel
+	for _, v := range lo.Models {
+		if v.Nonmem.Error != nil {
+			counter++
+			errors = append(errors, v.Nonmem)
+		}
+	}
+
+	if counter > 0 {
+		log.Printf("It appears that %d models generated an error during the initial setup phase", len(errors))
+		for _, v := range errors {
+			log.Printf("Model named %s has errored. Details: %s", v.Model, v.Error.Error())
+		}
+	}
+
+	//Models in OK state
+	log.Printf("%d models successfully completed initial setup phase.", len(lo.Models)-len(errors))
+
+	//Create signature safe slice for manager
+	var scalables []turnstile.Scalable
+
+	for _, v := range lo.Models {
+		scalables = append(scalables, v)
+	}
+
+	//Begin Execution
+	m := turnstile.NewManager(scalables, uint64(viper.GetInt("threads")))
+
+	now := time.Now()
+
+	go m.Execute()
+
+	//If we're in debug mode, let's periodically print out the details for the manager
+	if debug {
+		go func(m *turnstile.Manager) {
+			for {
+				log.Printf("Manager Details: Working: %d, Errors: %d, Completed: %d, Concurrency: %d, Iterations: %d", m.Working, m.Errors, m.Completed, m.Concurrency, m.Iterations)
+				time.Sleep(500 * time.Millisecond)
+			}
+		}(m)
+	}
+
+	//Basically wait
+	for !m.IsComplete() {
+		time.Sleep(5 * time.Millisecond)
+	}
+
+	if len(lo.Models) > 0 {
```

Comment:
above you're already checking if len(lo.Models) == 0 fatally log, so probably don't need to check again

Reply:
Yeah this is kind of a holdover I need to cleanup. It was there after completion during initial design to make sure I could capture "Nothing happened", but the earlier checks were added later. This can safely be removed.